Common/AR/Ar.md

```diff
@@ -0,0 +1,197 @@
+Sets the `AR_TOOLKIT_SOURCE_PATH` variable to the path of the sources of the toolkit.
+
+2. Installs the ArcGIS Runtime SDK for Qt version 100.6
+
+3. In Qt Creator, creates a new project "ArcGIS Runtime 100.6 Qt Quick C++ app" or "ArcGIS Runtime 100.6 Qt Quick QML app".
+
+4. In the projet file (.pro), includes the .pri file corresponding to the API used for the project:
+
+```
+include($$AR_TOOLKIT_SOURCE_PATH/Plugin/CppApi/ArCppApi.pri) // for C++ API
+```
+or
+```
+include($$AR_TOOLKIT_SOURCE_PATH/Plugin/QmlApi/ArQmlApi.pri) // for QML API
+```
+
+5. Updates the `Info.plist` file to request persmission for camera:
+
+```
+<key>NSCameraUsageDescription</key>
+<string>Camera access is needed for AR testing</string>
+```
+
+See [NSCameraUsageDescription Apple's documentation](https://developer.apple.com/documentation/bundleresources/information_property_list/nscamerausagedescription?language=objc) for details.
+
+6. The project is ready to be build and run in an iOS device. You need to verify the compatibility of your device
+with the ARKit.
+
+## ARCore installation
+
+For the details to uses `ARCore`, pleases refere to the [Google's documentation](https://developers.google.com/ar/).
+The details of the installation of the `ARCore` is describes in the
+[documentation](https://developers.google.com/ar/develop/c/enable-arcore).
+
+To create an application with the `ARCore` support:
+
+1. Downloads the sources of the [ArcGIS Runtime API Toolkit: https://github.com/Esri/arcgis-runtime-toolkit-qt.
+The `AR_TOOLKIT_SOURCE_PATH` variable refers to the path of the sources of the toolkit.
+
+2. Downloads and extracts the ARCore SDK for Android v1.10: https://github.com/google-ar/arcore-android-sdk/releases.
+This project contains the C header file (`arcore_c_api.h`) used to build the ARCore wrapper.
+The `AR_CORE_HEADER_PATH` variable refers to the path of the ARCore header files.
+
+3. Downloads the ARCore `aar` file from the Maven repository: https://maven.google.com/com/google/ar/core/1.10.0/core-1.10.0.aar
+This archive contains the native libraries used by the ARCore wrapper. Renames the extention file from `aar` to `zip` and
+extracts it.
+The `AR_CORE_BINARIES_PATH` variable refers to the path of the files of the ARCore native libraries.
+
+4. Installs the ArcGIS Runtime SDK for Qt version 100.6
+
+5. In Qt Creator, creates a new project "ArcGIS Runtime 100.6 Qt Quick C++ app" or "ArcGIS Runtime 100.6 Qt Quick QML app".
+
+6. In the projet file (.pro), includes the .pri file corresponding to the API used for the project:
+
+```
+include($$AR_TOOLKIT_SOURCE_PATH/ArCppApi.pri) // for C++ API
+```
+or
+```
+include($$AR_TOOLKIT_SOURCE_PATH/ArQmlApi.pri) // for QML API
+```
+
+7. Copy the gradle files from the ARCore SDK for Android to the `Android` dir in our project.
+
+8. In the Android manifest file `AndroidManifest.xml`, adds the following entries:
+
+```
+<meta-data android:name="com.google.ar.core" android:value="optional" />
```

Comment:
Good point. The `ArcGISArView` should work in the future without ARKit/ARCore (https://devtopia.esri.com/runtime/queen-annes-revenge/issues/154), using the sensors of the device. For the moment, this is not implemented.

With `required`, the app is visible in the app stores only for the devices which support AR. With `optional`, the app is visible for all the devices.

If the user want to create an app which contains an AR scene and a "normal" map/scene, then this parameter must be `optional`. If not (this is the case of the samples), `required` is fine.

I suppose I can improve the doc to explains that.

Reply:
Ok, that's good to know about the app store restrictions.

I think for now, we should advise people of the limitations as you've mentioned (regarding app store restrictions), but this doc should show/recommend `android:value="required"` since it will not function without that.

Reply:
I changed the value to `required` and add the following details:

```
The value can be `optional` or `required`. An AR optional app can be installed and run on devices
that don't support `ARCore` and the `Google Play Services for AR` is not automatically installed
with the app. An AR required app is available only on devices that support ARCore and the
`Google Play Services for AR` is automatically installed.

For details, see [Enable ARCore documentation](https://developers.google.com/ar/develop/java/enable-arcore#manifest).
```